mapper-runtime/src/main/java/com/datastax/oss/driver/api/mapper/entity/EntityHelper.java

```diff
@@ -188,6 +188,23 @@
   @NonNull
   Select selectByPrimaryKey();
 
+  /**
+   * Similar to {@link #selectByPrimaryKey()} but may operate over a subset of primary key
+   * components (partition key + clustering columns) to fetch an entity or entities.
+   *
+   * <p><code>parameterCount</code> components are listed in the {@code WHERE} clause as bindable
+   * values (the bind markers have the same names as the columns). They are listed in the natural
+   * order, i.e. partition key columns first, followed by clustering columns (in the order defined
+   * by the {@link PartitionKey} and {@link ClusteringColumn} annotations on the entity class).
+   *
+   * <p>If <code>parameterCount</code> of 0 is provided, this is the equivalent of a select query
+   * without a where clause, which queries for all rows in a table.
+   */
+  @NonNull
+  default Select selectByPrimaryKeyParts(int parameterCount) {
+    throw new UnsupportedOperationException("selectbyPrimaryKeyParts is unimplemented");
```

Comment:
Done to avoid breaking changes, although it is a bit awkward.

Reply:
> Done to avoid breaking changes, although it is a bit awkward.

Yes, and this only hides the breaking change by turning it into a runtime error. I've been thinking about other possible approaches:

1. introduce a sub-interface and require a cast (as we've done in other cases). The problem is that this doesn't scale well: we add `ExtendedProductHelper` now but what if we need more methods in the next version? `ExtendedExtendedProductHelper`?
2. Accept the breaking change and document it, on the assumption that clients probably don't implement this interface. The problem is in the "probably"; also, it's a slippery slope.
3. Don't add those methods to `EntityHelper`. Instead, generate the query directly from the method generators. For SELECT we already have `selectStart()`; for DELETE we have to duplicate the whole method, but it's not huge either.

    We won't make those queries available to users through the interface (e.g. for custom query providers), but they're not that hard to rewrite from scratch anyway.

My vote goes to 3.

Reply:
> Don't add those methods to EntityHelper

You know, it didn't occur to me that generated code used the implementations instead of declaring things as `EntityHelper`.  In this case, there is no reason to add something to the interface, so I agree this seems like a good course of action.

It looks like the only place (currently) that `EntityHelper` is used as an interface in other code is in `QueryProvider` and in `DaoBase`, where it's more used for setting/getting fields.

Since this method is used for query generation, I suppose it's probably fine to not put directly in `EntityHelper`.

But I do wonder if we should just consider removing all the query-generation based methods from `EntityHelper` now if we see no utility.  What do you think?

Reply:
> You know, it didn't occur to me that generated code used the implementations instead of declaring things as `EntityHelper`.

It didn't occur to me either 😆 I was initially thinking of moving the code to the DAO implementation class, but keeping them as "internal" methods in the helper class is even better.

> removing all the query-generation based methods from `EntityHelper`

As we discussed privately, those methods can be useful for `@QueryProvider` implementors, so it's nice to have them on the interface.